Sources/SwiftParser/Nominals.swift

```diff
@@ -339,6 +339,10 @@ extension Parser {
     )
   }
 
+  mutating func atInheritanceListTerminator() -> Bool {
+    return self.experimentalFeatures.contains(.trailingComma) && (self.at(.leftBrace) || self.at(.keyword(.where)))
+  }
```

Comment:
Am I missing something or do we still allow trailing commas in `associatedtype` declarations even though it’s ambiguous in the current state of your PR?

Reply:
I can't find a example that my implementation would allow trailing commas in `associatedtype`. Do you have you that you think it would?

Reply:
Indeed it would still parse in the following case because there's a `where` terminator:

```swift
protocol Foo {
    associatedtype T: P1, P2, where P1: Equatable, P2: Equatable
}
```